File: Features/referencestates-sanctionsbody/contracts/src/test/java/net/corda/samples/referencestates/states/SanctionableIOUStateTests.java
package net.corda.samples.referencestates.states;

import net.corda.core.contracts.ContractState;
import net.corda.core.contracts.LinearState;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.identity.CordaX500Name;
import net.corda.core.identity.Party;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import static org.jgroups.util.Util.assertEquals;
import static org.jgroups.util.Util.assertTrue;

public class SanctionableIOUStateTests {

    TestIdentity a = new TestIdentity(new CordaX500Name("Alice", "", "GB"));
    TestIdentity b = new TestIdentity(new CordaX500Name("Bob", "", "GB"));

    @Test
    public void constructorTest() {

        int value = 50;
        Party lender = a.getParty();
        Party borrower = b.getParty();
        UniqueIdentifier uid = new UniqueIdentifier();

        SanctionableIOUState st = new SanctionableIOUState(value, lender, borrower, uid);

        assertTrue(st instanceof ContractState);
        assertTrue(st instanceof LinearState);

        assertEquals(value, st.getValue());
        assertEquals(lender, st.getLender());
        assertEquals(borrower, st.getBorrower());
        assertEquals(uid, st.getLinearId());

        assertTrue(st.getParticipants().contains(a.getParty()));
        assertTrue(st.getParticipants().contains(b.getParty()));

    }

}

File: kotlin-samples/Features/referenceStates-sanctionsBody/contracts/src/test/kotlin/net/corda/samples/referencestates/states/SanctionableIOUStateTests.kt
package net.corda.samples.referencestates.states

import net.corda.core.contracts.ContractState
import net.corda.core.contracts.LinearState
import net.corda.core.contracts.UniqueIdentifier
import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import org.junit.Assert
import org.junit.Test

class SanctionableIOUStateTests {
    var a = TestIdentity(CordaX500Name("Alice", "", "GB"))
    var b = TestIdentity(CordaX500Name("Bob", "", "GB"))

    @Test
    fun constructorTest() {
        val value = 50
        val lender = a.party
        val borrower = b.party
        val uid = UniqueIdentifier()
        val st = SanctionableIOUState(value, lender, borrower, uid)
        Assert.assertTrue(st is ContractState)
        Assert.assertTrue(st is LinearState)
        Assert.assertEquals(value, st.value)
        Assert.assertEquals(lender, st.lender)
        Assert.assertEquals(borrower, st.borrower)
        Assert.assertEquals(uid, st.linearId)
        Assert.assertTrue(st.participants.contains(a.party))
        Assert.assertTrue(st.participants.contains(b.party))
    }
}
